lib/executor/base_executor.go

```diff
@@ -37,33 +37,37 @@ import (
 // inside of most of the executors, for the purpose of reducing boilerplate
 // code.
 type BaseExecutor struct {
-	config         lib.ExecutorConfig
-	executionState *lib.ExecutionState
-	VUIDLocal      *uint64 // counter for assigning executor-specific VU IDs
-	// Counter for keeping track of all VU iterations completed by this executor
-	// in the current (local) k6 instance.
-	iterLocal *uint64
-	logger    *logrus.Entry
-	progress  *pb.ProgressBar
+	config            lib.ExecutorConfig
+	executionState    *lib.ExecutionState
+	VUIDLocal         *uint64 // counter for assigning executor-specific VU IDs
+	iterationSegIndex *lib.SegmentedIndex
+	logger            *logrus.Entry
+	progress          *pb.ProgressBar
 }
 
 // NewBaseExecutor returns an initialized BaseExecutor
 func NewBaseExecutor(config lib.ExecutorConfig, es *lib.ExecutionState, logger *logrus.Entry) *BaseExecutor {
 	// Start at max uint64 so that the first iteration can be 0
-	startIterLocal := ^uint64(0)
+	start, offsets, lcd := es.ExecutionTuple.GetStripedOffsets()
+	segIdx := lib.NewSegmentedIndex(start, lcd, offsets)
 	return &BaseExecutor{
-		config:         config,
-		executionState: es,
-		VUIDLocal:      new(uint64),
-		iterLocal:      &startIterLocal,
-		logger:         logger,
+		config:            config,
+		executionState:    es,
+		VUIDLocal:         new(uint64),
+		logger:            logger,
+		iterationSegIndex: segIdx,
 		progress: pb.New(
 			pb.WithLeft(config.GetName),
 			pb.WithLogger(logger),
 		),
 	}
 }
 
+func (bs *BaseExecutor) NextIterationCounters() (uint64, uint64) {
```

Comment:
The linter caught it, but let's document what each return value means.

Also, this can be unexported?

Reply:
resolved in cfd3f59

Reply:
I'd expand more on that, but I can change it in #1836.